server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -431,10 +431,14 @@ public void connectToNode(final DiscoveryNode node, ConnectionProfile connection
     public ConnectionManager.ConnectionValidator connectionValidator(DiscoveryNode node) {
         return (newConnection, actualProfile, listener) -> {
             // We don't validate cluster names to allow for CCS connections.
-            handshake(newConnection, actualProfile.getHandshakeTimeout().millis(), cn -> true, ActionListener.map(listener, resp -> {
-                final DiscoveryNode remote = resp.discoveryNode;
-                if (node.equals(remote) == false) {
-                    throw new ConnectTransportException(node, "handshake failed. unexpected remote node " + remote);
+            handshake(newConnection, actualProfile.getHandshakeTimeout().millis(), cn -> true, ActionListener.map(listener, response -> {
+                final DiscoveryNode responseNode = response.discoveryNode;
+                /**
+                 * Validate if the node we are trying to connect to and the response recieved remote node is the same.
+                 * If not, error out.
+                 */
+                if (!node.equals(responseNode)) {
```

Comment:
Sorry, I didn't get the change - isn't it the same as `node.equals(remote) == false` ? (I think @nknize prefers explicit `false` checks)

Reply:
It felt more readable. 
But if we prefer the earlier, I am ok with it :) 

Thanks for the review @reta!

Reply:
Do you still feel the value for the comments ? 
If not I can close out the PR.

Reply:
To be fair, the previous version with `remote` was kind of explicit - the `node` should not be the same as `remote` one ...